src/lib/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -910,6 +911,50 @@ public function loadVersionInfo(int $contentId, ?int $versionNo = null): array
         return $queryBuilder->execute()->fetchAll(FetchMode::ASSOCIATIVE);
     }
 
+    public function loadVersionNoArchivedWithin(int $contentId, int $seconds): array
+    {
+        $now = new \DateTime();
+        $interval = DateInterval::createFromDateString($seconds . ' seconds');
+        if ($interval === false) {
+            return [];
+        }
+        $now->sub($interval);
```

Comment:
If you're not gonna use any `DateTime` functionality, then just go straight for the data you are after.
```suggestion
        $now = strtotime(sprintf('-%d seconds', $seconds));
```
(which also means you probably want to change the variable name)

And by the way, `new DateTime('-5 seconds')` is also a valid instantiation for `DateTime` objects.